src/Arguments/ArgumentsResolver.php

```diff
@@ -68,32 +68,34 @@ public static function parseDefinitionsByMethodSignature(
 
         $result = [];
         foreach ($parameters as $parameter) {
-            // match by position
-            if (array_key_exists($parameter->getPosition(), $definitions)) {
-                $result[] = static::parseDefinition($container, $definitions[$parameter->getPosition()]);
-                continue;
-            }
-            // match by name
-            if (array_key_exists($parameter->getName(), $definitions)) {
-                $result[] = static::parseDefinition($container, $definitions[$parameter->getName()]);
-                continue;
-            }
-            // autowiring
             $class = $parameter->getClass();
             $className = $class ? $class->getName() : null;
-            if ($class !== null && $container->has($className)) {
-                $result[] = new ContainerArgument($className);
-                continue;
-            }
-            // skip optional parameters
-            if ($parameter->isOptional()) {
-                $result[] = new ValueArgument($parameter->getDefaultValue());
-                continue;
+
+            switch (true) {
+                // match by position
+                case array_key_exists($parameter->getPosition(), $definitions):
+                    $result[] = static::parseDefinition(
+                        $container,
+                        $definitions[$parameter->getPosition()],
+                        (bool)$class
```

Comment:
I prefer to do this explicitly `$class !== ''`. But for doing this you should fix up line 72 like this:

```
$className = $class ? $class->getName() : '';
```

Reply:
Sorry, but this doesn't related to `$className`, this checks only `$class` is object, otherwise it will be always `null`